src/helpers/validator.ts

```diff
@@ -1,30 +1,39 @@
-import { Value } from "@sinclair/typebox/value";
-import { envSchema, envValidator, PluginSettings, pluginSettingsSchema, pluginSettingsValidator } from "../types";
+import { TransformDecodeCheckError, TransformDecodeError, Value, ValueError } from "@sinclair/typebox/value";
+import { Env, envSchema, envValidator, PluginSettings, pluginSettingsSchema, pluginSettingsValidator } from "../types";
 
-export function validateAndDecodeSchemas(env: object, rawSettings: object) {
+export function validateAndDecodeSchemas(rawEnv: object, rawSettings: object) {
+  const errors: ValueError[] = [];
+
+  console.log(rawEnv, rawSettings);
+  const env = Value.Default(envSchema, rawEnv) as Env;
   if (!envValidator.test(env)) {
-    const errors: object[] = [];
     for (const error of envValidator.errors(env)) {
-      const errorMessage = { path: error.path, message: error.message, value: error.value };
-      console.error(errorMessage);
-      errors.push(errorMessage);
+      console.error(error);
+      errors.push(error);
     }
-    throw new Error(`Invalid environment provided. ${errors}`);
   }
-  const envDecoded = Value.Decode(envSchema, env || {});
 
   const settings = Value.Default(pluginSettingsSchema, rawSettings) as PluginSettings;
   if (!pluginSettingsValidator.test(settings)) {
-    const errors: object[] = [];
     for (const error of pluginSettingsValidator.errors(settings)) {
-      const errorMessage = { path: error.path, message: error.message, value: error.value };
-      console.error(errorMessage);
-      errors.push(errorMessage);
+      console.error(error);
```

Comment:
What about our logger?

It seems to me that we spent all this time and money to implement it and then completely removed it. 

It was stable saving everything in Supabase, we even had log purge cron jobs every day.

Reply:
Leftover debug logs I forgot to remove, done.